pkg/runtime/path.go

```diff
@@ -0,0 +1,76 @@
+package runtime
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/alibaba/pouch/apis/types"
+)
+
+var (
+	runtimeDir                    = "runtimes"
+	runtimeDirPerm    os.FileMode = 0700
+	runtimeScriptPerm os.FileMode = 0700
+
+	runtimePaths map[string]string
+)
+
+// InitialPath keeps real runtime path in runtimePaths, runtimePaths only keep
+// runtime with path or args. If runtime.args passed, we will make a executable
+// script as a path, or runtime.path is record as path.
+// NOTE: containerd not support runtime args directly, so we make executable
+// script include runtime path and args as a runtime execute binary.
+func InitialPath(baseDir string, runtimes map[string]types.Runtime) error {
+	// initializes map keep runtime script path
+	if runtimePaths == nil {
+		runtimePaths = make(map[string]string)
+	}
+
+	dir := filepath.Join(baseDir, runtimeDir)
+
+	// remove runtime scripts last generated, since runtime config may changed
+	// every time daemon start.
+	if err := os.RemoveAll(dir); err != nil {
+		return fmt.Errorf("failed to clean runtime scripts directory %s: %s", dir, err)
+	}
+
+	if err := os.MkdirAll(dir, runtimeDirPerm); err != nil {
+		return fmt.Errorf("failed to new runtime scripts directory %s: %s", dir, err)
+	}
+
+	// create script for runtime who has args
+	for name, r := range runtimes {
+		if len(r.RuntimeArgs) == 0 {
+			if r.Path != "" {
+				runtimePaths[name] = r.Path
+			}
+			continue
+		}
+
+		script := filepath.Join(dir, name)
+		if r.Path == "" {
+			r.Path = name
+		}
+		data := fmt.Sprintf("#!/bin/sh\n%s %s $@\n", r.Path, strings.Join(r.RuntimeArgs, " "))
+
+		if err := ioutil.WriteFile(script, []byte(data), runtimeScriptPerm); err != nil {
+			return fmt.Errorf("failed to create runtime script %s: %s", script, err)
```

Comment:
In this place you directly return the error, while some data in pouchd has been changed, for example, `runtimePaths`.
In this case, do we need to revert the change of `runtimePaths `? just double check.

Reply:
if fails here, daemon won't start.

Reply:
> if fails here, daemon won't start.

Oh, that quite makes sense. Thanks.